File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        nmst=set(nums)
        lgst=0
        for n in nmst:
            if n- 1 not in nmst:
                curr=n
                l=1
                while curr+1 in nmst:
                    curr+=1
                    l+=1
                lgst= max(lgst,l)
        return lgst

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums){
        unordered_set<int>nmst(nums.begin(), nums.end());
        int lgst=0;
        for(int n:nmst){
            if(nmst.find(n-1)==nmst.end()){
                int curr=n;
                int l=1;
                while(nmst.find(curr+1)!=nmst.end()){
                    curr+=1;
                    l+=1;
                }
                lgst=max(lgst,l);
            }
        }
        return lgst;
    }
};